src/os/bluestore/BlueFS.cc

```diff
@@ -1556,7 +1556,10 @@ int BlueFS::_read_random(
       s_lock.unlock();
       uint64_t x_off = 0;
       auto p = h->file->fnode.seek(off, &x_off);
-      uint64_t l = std::min(p->length - x_off, static_cast<uint64_t>(len));
+      ceph_assert(p != h->file->fnode.extents.end());
+      uint64_t l = std::min(p->length - x_off, len);
```

Comment:
actually this has got modifications from multiple PRs. Original one you're backporting includes just ceph_assert()